runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcClickboxOverlay.java

```diff
@@ -80,8 +79,19 @@ public Dimension render(Graphics2D graphics)
 
 	private void renderNpcOverlay(Graphics2D graphics, NPC actor, String name, Color color)
 	{
+		Polygon objectTile = actor.getCanvasTilePoly();
 		Polygon objectClickbox = actor.getConvexHull();
```

Comment:
This would perform a bit better if you rearranged this code so that you only get these Polygons if you need them. For example, only call `actor.getCanvasTilePoly()` if `config.renderStyle() == RenderStyle.TILE`.